java/core/src/main/java/com/microsoft/bond/ListBondType.java

```diff
@@ -98,13 +107,14 @@ protected final void serializeValue(SerializationContext context, List<TElement>
         this.verifyNonNullableValueIsNotSetToNull(value);
         int count = value.size();
         context.writer.writeContainerBegin(count, this.elementType.getBondDataType());
-        for (int i = 0; i < count; ++i) {
+        int i = 0;
+        for (TElement element : value) {
```

Comment:
I noticed an inefficient O(N^2) implementation here, that was copy-paste from the VectorBondType class. I'd like to keep it as is, since the change is small and important for perf.

---
In reply to: [128378679](https://github.com/Microsoft/bond/pull/532#discussion_r128378679) [](ancestors = 128378679)